src/prefect/tasks/monte_carlo/monte_carlo_lineage.py

```diff
@@ -348,28 +353,43 @@ def run(
             raise ValueError("Value for `object_type` must be provided.")
         if resource_name is None:
             raise ValueError("Value for `resource_name` must be provided.")
-        if metadata_key is None:
-            raise ValueError("Value for `metadata_key` must be provided.")
-        if metadata_value is None:
-            raise ValueError("Value for `metadata_value` must be provided.")
         if api_key_id is None:
             raise ValueError("Value for `api_key_id` must be provided.")
         if api_token is None:
             raise ValueError("Value for `api_token` must be provided.")
 
         mc = MonteCarloClient(api_key_id, api_token)
-        node_mcon = mc.create_or_update_lineage_node_with_tag(
-            node_name,
-            object_id,
-            object_type,
-            resource_name,
-            metadata_key,
-            metadata_value,
-        )
-        if prefect_context_tag:
+        if lineage_tags is None:
+            node_mcon = mc.create_or_update_lineage_node(
+                node_name,
+                object_id,
+                object_type,
+                resource_name,
+            )
+        else:
+            for tag in lineage_tags:
+                if sorted(tag.keys()) != ["propertyName", "propertyValue"]:
+                    raise MonteCarloIncorrectTagsFormat(
+                        "Must provide tags in the format "
+                        '[{"propertyName": "tag_name", "propertyValue": "tag_value"}].',
+                        "You provided: ",
+                        tag,
+                    )
+            node_mcon = mc.create_or_update_lineage_node_with_multiple_tags(
+                node_name,
+                object_id,
+                object_type,
+                resource_name,
+                lineage_tags,
+            )
+        if prefect_context_tags:
             self.logger.info("Setting Prefect context tags on the lineage nodes...")
-            mc.create_or_update_tags_for_mcon(
-                mcon=node_mcon, key="prefect_context", value=get_prefect_context()
+            mc.create_or_update_lineage_node_with_multiple_tags(
+                node_name,
+                object_id,
+                object_type,
+                resource_name,
+                tags=get_prefect_context_tags(),
```

Comment:
Now that we have a method to set multiple tags at once in MC, we can make only one call for the user defined tags and the prefect context tags.

Reply:
you're right, I've just pushed a change. Thanks for that!